x-pack/solutions/observability/plugins/apm/server/routes/environments/get_environments.ts

```diff
@@ -53,13 +54,21 @@ export async function getEnvironments({
       environments: {
         terms: {
           field: SERVICE_ENVIRONMENT,
-          missing: ENVIRONMENT_NOT_DEFINED.value,
           size,
         },
       },
     },
   };
 
+  const hasUnsetEnvironments = hasUnsetValueForField({
+    apmEventClient,
+    searchAggregatedTransactions,
+    serviceName,
+    fieldName: SERVICE_ENVIRONMENT,
+    start,
+    end,
+  });
+
   const resp = await apmEventClient.search(operationName, params);
```

Comment:
You could run these in parallel with Promise.all

Reply:
wouldn't adding a Promise.all block until both were completed? Maybe I'm not understanding completely how async works in JS, but wouldn't this configuration allow `hasUnsetEnvironments` to be running in the background, or is js only optimized to parallelize asnyc tasks within a promise.all?

Reply:
It would. I'm not sure if there is any significant performance gain here because this endpoint needs both results before it responds. `hasUnsetValueForField` is not a heavy call that could benefit from being deferred.

Start early, await later is still a valid pattern, so it's up to you.